### DISCUSSION TITLE
specify force to apply from one object onto another

### DISCUSSION THREAD

**Discussion by shivanimall** [bulletphysics/bullet3#3853] [Q&A]
Hello @erwincoumans,

Say I have object A and a robot B. I am using robot B to apply force onto object A at point C.
Is it possible to control/specify the exact amount of force robot B should apply onto object A (say at a contact point C)? 

For example: I want robot B's finger (loaded with a sensor) to apply 3 N force on object A at contact point (0, 0.1, 0.5). (I am fine with small error threshold of <= 0.2)

I have looked at applyExternalForce and setJointControlArray(TORQUE_CONTROL), these don't exactly seem to serve my purpose, but I may have missed something, let me know. My codebase is in pybullet but I am open trying out other suggestions/libraries you have for this.

Thank you, please let me know if my question is unclear.

**Comment by erwincoumans** [comment]
A robot uses actuators (motors) to move end-effectors (fingers) that apply forces, so it is best to measure the contact forces, and increase/decrease the motor torque to reach desired contact forces.

Also, there is a chance someone else answers, so no need to add my name ;-)

**Reply by shivanimall** [reply]
thank you Erwin :) yes, sounds good.
